src/DssBatchedEmergencySpell.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-FileCopyrightText: © 2024 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity ^0.8.16;
+
+import {DssEmergencySpell, DssEmergencySpellLike} from "./DssEmergencySpell.sol";
+
+interface DssBatchedEmergencySpellLike is DssEmergencySpellLike {
+    function ilks() external view returns (bytes32[] memory);
+}
+
+/// @title Batched Emergency Spell
+/// @notice Defines the base implementation for batched emergency spells.
+/// @custom:authors [amusingaxl]
+/// @custom:reviewers []
+/// @custom:auditors []
+/// @custom:bounties []
+abstract contract DssBatchedEmergencySpell is DssEmergencySpell, DssBatchedEmergencySpellLike {
+    /// @dev The min size for the list of ilks
+    uint256 public constant MIN_ILKS = 2;
+    /// @dev The max size for the list of ilks
+    uint256 public constant MAX_ILKS = 3;
```

Comment:
Technically, nothing prevents us from setting `MIN_ILKS = 1` and replacing the single-ilk spells with this one. We'd just need to pass a list containing 1 single ilk to the constructor.